### PR TITLE
Add bang variants

### PR DESCRIPTION
**Pull request by dbernheisel**
This adds bang variants of the parse functions, 
* `parse!`
* `parse_datetime!`
* `parse_date!`
* `parse_time!`

When parsing fails, it will raise a `DateTimeParser.ParseError` with the `message` key populated.

### CODE REVIEW

File: lib/date_time_parser.ex [hunk 1]
```diff
@@ -164,6 +164,20 @@ defmodule DateTimeParser do
     end
   end
 
+  @doc """
+  Parse a `%DateTime{}`, `%NaiveDateTime{}`, `%Date{}`, or `%Time{}` from a string.
+
```
Comment (tmr08c) [inline lib/date_time_parser.ex]: Any reason this one has the extra newline and the others don't?

File: lib/error.ex [hunk 2]
```diff
@@ -0,0 +1,5 @@
+defmodule DateTimeParser.Error do
```
Comment (tmr08c) [inline lib/error.ex]: Is it overkill to make this `ParseError` to be more specific?

Comment (dbernheisel) [reply lib/error.ex]: I couldn't think of anything better on the spot :) but yea if you have a suggestion I'll be happy to put it in.

File: lib/date_time_parser.ex (hunk 1 above)
Comment (dbernheisel) [reply lib/date_time_parser.ex]: no intentional reason. I'll double-check the `mix docs` output and make it consistent either way. Nice catch.

Comment (dbernheisel) [reply lib/date_time_parser.ex]: Updated to be consistent. I put the sentence about "accepting x options" on a newline for all functions.

File: lib/error.ex (hunk 2 above)
Comment (tmr08c) [reply lib/error.ex]: I made #26 so you can see if it looks like a good rename

**Pull request merged**